### ISSUE TITLE
support vi ft comments

### ISSUE THREAD

**Issue by xenoterracide** [sharkdp/bat#1799]
Sadly, not all file types encode their file type as an extension. Even worse, a lot of files use meaningless extensions where you have to look at their directory. For example, systemd uses `environment.d/*.conf` for `.env` files, that could also be parsed as `.sh` because they also allow for simple interpolation.

Would be cool to support the vi way for being able to use a comment to define it.  Suggesting specifically this because many files may already have it, and/or it serves another purpose.

https://unix.stackexchange.com/a/19869/29

e.g.

```
# vi: ft=gitconfig
```

**Comment by sharkdp** [comment]
This has been requested a few times. Please see #1732, for example.

> Even worse, a lot of files use meaningless extensions where you have to look at their directory. For example, systemd uses `environment.d/*.conf` for `.env` files, that could also be parsed as `.sh` because they also allow for simple interpolation.

You can actually tell `bat` to map paths like this to a specific syntax. For your example, you would have to add something like
```bash
--map-syntax '**/environment.d/*.conf:Bourne Again Shell (bash)'
```
to `bat`s config file.

**Comment by sharkdp** [comment]
`bat` actually highlights these systemd files with the INI syntax currently (see https://github.com/sharkdp/bat/pull/1060). Let us know if you think this can be improved.

**Comment by xenoterracide** [comment]
![Screenshot from 2021-08-16 17-58-12](https://user-images.githubusercontent.com/5517/129634691-a76b072a-59f2-4228-900c-3e0acb486bc4.png)
looks pretty flat on my system, ;) and they're closer to sh than INI, afaik. Obviously the're an example. I've seen some `.conf` files recently that I'm not even sure what they are...

**Comment by SuperSandro2000** [comment]
> --map-syntax '**/environment.d/*.conf:Bourne Again Shell (bash)'

Could we shorten this to ``--map-syntax '**/environment.d/*.conf:bash'`` ?

**Comment by Enselic** [comment]
Duplicate of #160

... which is closed. It maybe should be reopened.

Either way, this is a duplicate of that one.

**Comment by Enselic** [comment]
Duplicate of #160
